agent/token_store.go

```diff
@@ -0,0 +1,84 @@
+package agent
+
+import (
+	"sync"
+)
+
+// TokenStore is used to hold the special ACL tokens used by Consul agents. It
+// is designed to update the tokens on the fly, so the token store itself should
+// be plumbed around and used to get tokens at runtime, don't save the resulting
+// tokens.
+type TokenStore struct {
```

Comment:
Thought about that but this feels pretty tied to the agent itself; that would basically just be enforcing the rights on the unexported variables here which didn't feel worth the extra include. I think the bigger win is when we don't have access to the config everywhere we then will know someone can grab a potentially stale token from there.